packages/backend/src/apps/lettersg/dynamic-data/get-template-ids.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  DynamicDataOutput,
+  IDynamicData,
+  IGlobalVariable,
+} from '@plumber/types'
+
+import { Template } from './types'
+
+const dynamicData: IDynamicData = {
+  key: 'getTemplateIds',
+  name: 'Get Template IDs',
+  async run($: IGlobalVariable): Promise<DynamicDataOutput> {
+    try {
+      const { data } = await $.http.get('/v1/templates')
+
+      if (!data.templates) {
+        return {
+          data: [],
+        }
+      }
+
+      // map to dynamic data format which is { name: string; value: string }
+      return {
+        data: data.templates.map((template: Template) => ({
+          name: template.name,
+          value: template.templateId.toString(),
+        })),
+      }
+    } catch (error) {
+      return {
+        data: [],
+        error: error.message,
+      }
+    }
```

Comment:
most likely the `catch` block here will be triggered by failure in `$.http.get('/v1/templates')` - should we add logging so that we can investigate why that might fail?